app/src/main/java/org/schabi/newpipe/local/feed/FeedFragment.kt

```diff
@@ -243,18 +246,43 @@ class FeedFragment : BaseStateFragment<FeedState>() {
                 .show()
             return true
         } else if (item.itemId == R.id.menu_item_feed_toggle_played_items) {
-            showPlayedItems = !item.isChecked
-            updateTogglePlayedItemsButton(item)
-            viewModel.togglePlayedItems(showPlayedItems)
+            showStreamVisibilityDialog()
+        }
+
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun showStreamVisibilityDialog() {
+        val dialogItems = arrayOf(
+            getString(R.string.feed_show_watched),
+            getString(R.string.feed_show_partially_watched),
+            getString(R.string.feed_show_upcoming)
+        )
+
+        val checkedDialogItems = booleanArrayOf(!showPlayedItems, !showPartiallyPlayedItems, !showFutureItems)
```

Comment:
Mmmh, I think it's better for the dialog to be positive instead of negative (i.e. "Show the following streams"), to create less confusion. A checked checkbox would mean that the item is enabled.